x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_edit/fields/hooks/use_data_view.ts

```diff
@@ -30,6 +30,7 @@ export function useDataView(indexPatternsOrDataViewId: UseDataViewParams): UseDa
 
     (async () => {
       if (dataView !== undefined) {
+        setIsLoading(false);
         return;
       }
```

Comment:
I haven't tested the PR locally yet, but a quick suggestion is to move the return statement outside the async function, before setting the loading state (line 29). This would be more optimal, as it avoids setting the loading state and then immediately unsetting it in the next cycle, which can lead to wasted re-renders and potentially jittery UI.

Reply:
Yeah that makes sense, good idea

Reply:
Implemented in [55d33ff](https://github.com/elastic/kibana/pull/231699/commits/55d33ff1a1b3f5351b0749c797975f0d7ee04d17)